dlpy/network.py

```diff
@@ -173,6 +173,35 @@ def compile(self):
         if rt.severity > 1:
             raise DLPyError('cannot build model, there seems to be a problem.')
         self.num_params = 0
+
+        # before addLayer, self.layers are reordered based on layer_id
+        # the new id is created based on model.summary which contains layer_id.
+        # since it is not compiled, it might contains None or index from it original model summary.
+        orig_ids = self.summary['Layer Id']
```

Comment:
we have a layer id but somewhere I normally take it from the model info. to do that, we need to build the model. have you considered that?

Reply:
I just checked our code regarding layer id. We have a function called __load_layer_ids() which is used to get reliable layer id from a model table after compile(). Since __load_layer_ids is invoked in print_summary(), Functional model's layer id is taken from the model info as well. 

My fix actually solves a bug that happens when a predefined model is loaded along with its weights and I want to convert it to a functional model. Since cas assigns layer id in a different way from functional API, the fix rearranges the sequence of List layers to be called by addlayer action to ensure that the layer id keeps consistent after conversion. Moreover, the id mentioned here doesn't refer to the model layer id which will determine by __load_layer_ids(). So I believe it doesn't relate to the issue.